File: ALifeCommon/src/main/java/com/cristiano/alife/instructions/logic/OrInstruction.java
package com.cristiano.alife.instructions.logic;

import com.cristiano.alife.consts.ALifeConsts;
import com.cristiano.alife.world.BaseInstruction;
import com.cristiano.alife.world.IOrganismo;
import com.cristiano.alife.world.IWorld;

public class OrInstruction extends BaseInstruction  {
	public OrInstruction(IWorld mundo) {
		super(mundo);
	}
	@Override
	public void executa(IOrganismo o) {
		int v=o.getReg(getByteOrganismo(o,1));
		v^=1;
		o.setReg(getByteOrganismo(o,1), v);
	}
	
	@Override
	public String getDescription(IOrganismo o,int ip){
		return comment(name+" "+ALifeConsts.getLetter(o.getMemory(ip+1)),ALifeConsts.getLetter(o.getMemory(ip+1))+" <- or("+ALifeConsts.getLetter(o.getMemory(ip+1))+") ");
	}
	
	@Override
	public int getStep() {
		return 2;
	}

}

File: netAvida/netAvida.Backend/instructions/impl/logic/OrInstruction.cs
using netAvida.backend;
using netAvida.backend.interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace netAvida.Backend.instructions.impl
{
    public class OrInstruction : BaseInstruction
    {
        public OrInstruction(IWorld mundo): base(mundo)
        {
        }

    public override void executa(IOrganismo o)
        {
            int v = o.getReg(getByteOrganismo(o, 1));
            v ^= 1;
            o.setReg(getByteOrganismo(o, 1), v);
        }

    public override int getStep()
        {
            return 2;
        }

    public override String getDescription(IOrganismo o, int ip)
        {
            return comment(name + " " + ALifeConsts.getLetter(o.getMemory(ip + 1)), ALifeConsts.getLetter(o.getMemory(ip + 1)) + " <- or(" + ALifeConsts.getLetter(o.getMemory(ip + 1)) + ") ");
        }
    }
}
